docs/user_guide.rst

```diff
@@ -112,33 +112,42 @@ and then call execution methods without specifying additional keyword arguments.
 their current values using `backend.options`. Below table summarizes currently available options:
 
 .. list-table::
-   :widths: 25 20 25 100
+   :widths: 25 100
    :header-rows: 1
 
    * - Name
-     - Type
-     - Example value
      - Description
    * - `shots`
-     - int
-     - 1207
-     - Number of shots.
+     - Type: ``int``, Example value: ``1207``.
+
+       Number of shots.
    * - `calibration_set_id`
-     - str
-     - "f7d9642e-b0ca-4f2d-af2a-30195bd7a76d"
-     - Indicates the calibration set to use. Defaults to `None`, which means the IQM server will use the best
+     - Type: ``str``, Example value ``"f7d9642e-b0ca-4f2d-af2a-30195bd7a76d"``.
+
+       Indicates the calibration set to use. Defaults to `None`, which means the IQM server will use the best
        available calibration set automatically.
    * - `circuit_duration_check`
-     - bool
-     - False
-     - Enable or disable server-side circuit duration checks. The default value is `True`, which means if any job is
+     - Type: ``bool``, Example value: ``False``.
+
+       Enable or disable server-side circuit duration checks. The default value is `True`, which means if any job is
        estimated to take unreasonably long compared to the coherence times of the qubits, or too long in wall-clock
        time, the server will reject it. This option can be used to disable this behaviour. In normal use, the
        circuit duration check should always remain enabled.
    * - `heralding_mode`
-     - :py:class:`~iqm_client.iqm_client.HeraldingMode`
-     - "zeros"
-     - Heralding mode to use during execution. The default value is "none".
+     - Type: :py:class:`~iqm.iqm_client.iqm_client.HeraldingMode`, Example value: ``"zeros"``.
+
+       Heralding mode to use during execution. The default value is "none".
+   * - `circuit_callback`
+     - Type: :py:class:`collections.abc.Callable`, Example value: ``None``.
+
+       A function that accepts a list of :py:class:`qiskit.QuantumCircuit` instances and does not return anything.
```

Comment:
```suggestion
     - Type: :class:`collections.abc.Callable`, Example value: ``None``.

       A function that accepts a list of :class:`qiskit.QuantumCircuit` instances and does not return anything.
```

Reply:
So the idea is that `circuit_callback` can modify the given circuits in place?
Should it instead return the circuits to be executed as `list[qiskit.QuantumCircuit]`, so it can either return what it was given as is, or edited copies?

Reply:
No, it is not intended for modifying circuits. Its main intention is to extract some info that user needs from circuits, i.e. only read the circuits. As a side effect they can also modify circuits in-place, but that's not the intended use for this option.

Reply:
OK, maybe just mention in the docstring that you're allowed to modify the circuits in place.